ledfx/effects/spotlight.py

```
@@ -0,0 +1,514 @@
+            return
+        count = min(count, self.spot_count)
+        self.spot_head = (self.spot_head + count) % self.spot_capacity
+        self.spot_count -= count
+
+    def _pick_spot_center(self):
+        """Pick a random spotlight center, preferring spacing from active spots."""
+        if self.pixel_count <= 1:
+            return 0
+
+        if self.spot_count == 0:
+            return random.randrange(self.pixel_count)
```

Ruff (0.15.10)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)